### PR TITLE
Add warning to install widget for M1 GPUs

### PR DESCRIPTION
**Pull request by polm**
## Description

Currently support for M1 GPUs is experimental and limited, but the install widget allows you to select the M1/GPU combination as though everything will work. This adds a warning about the experimental status and links to the FAQ on the subject. This should prevent issues like #11641.

The way this displays is not ideal - for example, there's a link but it's not obvious because it's not underlined or anything. I will see if there's a way to modify the CSS to make it clearer, but any suggestions on other places to display this would also be welcome.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: website/src/widgets/quickstart-install.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ const QuickstartInstall = ({ id, title }) => {
                         showDropdown={showDropdown}
                     >
                         <QS os="mac" hardware="gpu" platform="arm">
-                            # Note M1 GPUs support is experimental, see <a href="https://github.com/explosion/spaCy/discussions/11436">#11436</a>
+                            # Note M1 GPUs support is experimental, see <a href="https://github.com/explosion/thinc/issues/792">this issue</a>
```
Comment (adrianeboyd) [inline website/src/widgets/quickstart-install.js:163]:
```suggestion
                            # Note M1 GPU support is experimental, see <a href="https://github.com/explosion/thinc/issues/792">this issue</a>
```

**Comment by polm** [comment]
I figured out how to modify the CSS to underline the URL. I don't think this affects any other documentation, or if it does I can't imagine it would be an issue, so I think this is OK?

**Comment by adrianeboyd** [comment]
While we're looking at this, can you take `conda install -c conda-forge cupy` out of the M1+GPU options?

File: website/src/widgets/quickstart-install.js [hunk 2]
```diff
@@ -159,6 +159,9 @@ const QuickstartInstall = ({ id, title }) => {
                         setters={setters}
                         showDropdown={showDropdown}
                     >
+                        <QS os="mac" hardware="gpu" platform="arm">
+                            # Note M1 GPU support is experimental, see <a href="https://github.com/explosion/thinc/issues/792">this issue</a>
```
Comment (adrianeboyd) [inline website/src/widgets/quickstart-install.js:163]: I'd like link text looks more informative than basically "here", but I'm not particularly attached to this particular suggestion:

```suggestion
                            # Note M1 GPU support is experimental, see <a href="https://github.com/explosion/thinc/issues/792">Thinc issue #792</a>
```

Comment (polm) [reply website/src/widgets/quickstart-install.js:163]: Looks good to me!

**Comment by polm** [comment]
Removed `cupy` from the M1+GPU options.

**Pull request merged**